Local minimizer table

wathen opened the pull request:
#### Description of Work

Fixes #401 

Adds an additional table which indicates whether the software converges to a local minimum


#### Testing Instructions

1. Check table is consistent.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

tyronerees commented on fitbenchmarking/utils/fitbm_result.py:
```diff
@@ -4,6 +4,8 @@
 
 import numpy as np
 
+LOCAL_MIN_TOL = 1e-2
```
I wonder if even this is too tight?  For example, the local minima for the gaussian tests are coming up as non local minima.  I think we should raise this to `1e-1`, but we also need to make sure it's clear to the user what we're telling them here (although the same is true for all the tables, so I don't think we need to do that here).

wathen replied:
I've set the tolerance to be `1e-1` but not reported this to the users yet. As you said, more information about the tables should be added but I don't think this is the place to do that

wathen commented:
> The code looks neater without the if statements, but the UI seems to have taken a step backwards in going from this:
> ![old_layout](https://user-images.githubusercontent.com/13645545/73062492-6a6f9180-3e94-11ea-868b-a74666506e78.png)
> to this:
> ![new_layout](https://user-images.githubusercontent.com/13645545/73062497-6e9baf00-3e94-11ea-942c-1eb357708e55.png)
> 
> Can we go back to the same formatting we had before? Ideally with the combined tables at the top again, as I'm sure that gets used most.

I have reverted the ordering back to what it was before and removed the indentation of the buttons.

wathen commented on fitbenchmarking/core/results_output.py:
```diff
@@ -15,11 +15,24 @@
+    "The local min results show whether the software has converged to a " \
+    " local minimum."
+
+SORTED_TABLE_LINKS = ["compare", "acc", "runtime", "local_min"]
```
```suggestion
SORTED_TABLE_NAMES = ["compare", "acc", "runtime", "local_min"]
```

tyronerees commented on fitbenchmarking/utils/fitbm_result.py:
```diff
@@ -68,6 +70,33 @@ def __init__(self, options=None, problem=None, fit_status=None,
+            norm_r = np.linalg.norm(r)
+            norm_min_test = np.linalg.norm(min_test)
+            self.norm_rel = norm_min_test / norm_r
+            if norm_r <= LOCAL_MIN_TOL or norm_min_test <= LOCAL_MIN_TOL \
```
Following on from the discussion friday, can we set two tolerances, `GRAD_TOL` and `RES_TOL`, and test `norm_r` with `RES_TOL` and the other two with `GRAD_TOL`.  We can keep the tolerance to `GRAD_TOL` as is, but tighten `RES_TOL` to, say, `1e-8`.

wathen replied:
Done

wathen commented:
> I still prefer the original titles on the index page sections -- "Combined accuracy and runtime results" is much more informative than "compare table". Was your intention to to link this with label the options the user sends in? If so, I think we should highlight that somewhere (in the docs?)
> 
> The description seemed more natural to me when it came before the buttons -- there were clear sections then, which are a little disjointed now.

I was going to do the docs as part of the refactoring but not here since we have changed the layout completely
